trin-core/src/utp/packets.rs

```diff
@@ -84,26 +89,7 @@ fn check_extensions(data: &[u8]) -> Result<(), &'static str> {
     Ok(())
 }
 
-impl<'a> TryFrom<&'a [u8]> for Packet {
-    type Error = &'static str;
-
-    // todo: Added support to check validity of extensions,
-    // we don't need to worry about checking payloads
-    fn try_from(value: &[u8]) -> Result<Self, Self::Error> {
-        match PacketHeader::try_from(value).and(check_extensions(value)) {
-            Ok(_) => Ok(Packet(value.to_owned())),
-            Err(e) => Err(e),
-        }
-    }
-}
-
-impl Default for PacketHeader {
-    fn default() -> Self {
-        Self::new()
-    }
-}
-
-#[derive(Debug)]
+#[repr(C)]
```

Comment:
Maybe a one-liner explaining why we want/need this? I'm seeing articles about it being useful for FFI, but I don't know  about plans to use that.

Reply:
Here is a more detailed explanation of [repr(C)](https://doc.rust-lang.org/reference/type-layout.html#the-c-representation). 
Default Rust representation doesn't guarantee the data layout and because we want to cast raw pointers to `PacketHeader`, we need consistent data layout representation of the struct.